### PR TITLE
Adds group_by_hostgroups parameter to Icinga2 inventory

### PR DESCRIPTION
**Pull request by Gianlu**
##### SUMMARY
This PR adds an optional parameter to Icinga2 inventory `group_by_hostgroups` (boolean). If `true`, Icinga2 hostgroups are used as Ansible groups; if `false` the step is ignored. For backward compatibility, the default is `true`.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

icinga2.py inventory

##### ADDITIONAL INFORMATION
In my use case, I want to use Icinga2 as inventory source but my hostgroups' naming convention isn't suitable to become an ansible group strategy (spaces, capital letters and invalid characters) so I prefer to disable the builtin grouping by hostgroups and use the capabilities of constructed inventory (groups and keyed_groups).

```YAML
# my.icinga2.yml
plugin: community.general.icinga2
url: http://localhost:5665
user: ansible
password: secure
host_filter: \"linux-servers\" in host.groups
validate_certs: false  # only do this when connecting to localhost!
group_by_hostgroups: false
```

### CODE REVIEW

File: changelogs/fragments/7998-icinga2-inventory-group_by_hostgroups-parameter.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - icinga2 inventory plugin - adds new parameter ``group_by_hostgroups`` in order to make grouping by Icinga2 hostgroups optional
```
Comment (felixfontein) [inline changelogs/fragments/7998-icinga2-inventory-group_by_hostgroups-parameter.yml:2]:
```suggestion
  - icinga2 inventory plugin - adds new parameter ``group_by_hostgroups`` in order to make grouping by Icinga2 hostgroups optional (https://github.com/ansible-collections/community.general/pull/7998).
```

File: plugins/inventory/icinga2.py [hunk 2]
```diff
@@ -63,6 +63,12 @@
         default: address
         choices: ['name', 'display_name', 'address']
         version_added: 4.2.0
+      group_by_hostgroups:
+        description:
+          - Uses Icinga2 hostgroups as groups
```
Comment (felixfontein) [inline plugins/inventory/icinga2.py:68]:
```suggestion
          - Uses Icinga2 hostgroups as groups.
```

File: plugins/inventory/icinga2.py [hunk 3]
```diff
@@ -248,12 +255,13 @@ def _convert_inv(self, json_data):
                 host_attrs['state'] = 'on'
             else:
                 host_attrs['state'] = 'off'
-            host_groups = host_attrs.get('groups')
             self.inventory.add_host(host_name)
-            for group in host_groups:
-                if group not in self.inventory.groups.keys():
-                    self.inventory.add_group(group)
-                self.inventory.add_child(group, host_name)
+            if (self.group_by_hostgroups):
```
Comment (felixfontein) [inline plugins/inventory/icinga2.py:259]:
```suggestion
            if self.group_by_hostgroups:
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

**Comment by felixfontein** [comment]
The change looks good, but the commit history looks quite ... adventerous :) Would you mind cleaning it up a bit? Thanks!

**Comment by felixfontein** [comment]
@Gianlu ping

**Comment by Gianlu** [comment]
Hi,
You are right: I didn't reply. Have I to squash? Or do you have a guide line I can follow?

Thanks

**Comment by felixfontein** [comment]
In general squashing is not necessary since we squash on merge, but all your commit messages will end up in the PR's commit message, which currently would contain something like:
```
* Introduce 'group_by_hostgroups' parameter

* Added CHANGELOG fragment

* Rename changelog fragmment in order to include PR number

* Linting after CI failure

* Linting after CI failure (2)

* Linting after CI failure (3)

* Fix test

* Fixing tests (2)

* Update plugins/inventory/icinga2.py

* Update changelogs/fragments/7998-icinga2-inventory-group_by_hostgroup…

* Update plugins/inventory/icinga2.py

* Linting after CI failure

* Update plugins/inventory/icinga2.py

* Error due to rebase

* Introduce 'group_by_hostgroups' parameter

* Added CHANGELOG fragment

* Rename changelog fragmment in order to include PR number

* Linting after CI failure

* Linting after CI failure (2)

* Linting after CI failure (3)

* Update plugins/inventory/icinga2.py

* Update changelogs/fragments/7998-icinga2-inventory-group_by_hostgroup…

* Linting after CI failure

* Linting after CI failure (2)

* Update plugins/inventory/icinga2.py

* Update plugins/inventory/icinga2.py

* Error in merging

* Correct over indent

* Fragment

* Deleted temp file
```
I would squash at least some of them together (or even all if you don't want to invest much time in it).

**Comment by felixfontein** [comment]
@Gianlu thanks for your contribution.

**Pull request merged**